src/update_halo.jl

```diff
@@ -477,8 +477,8 @@ end
 let
     global iwrite_sendbufs!, allocate_rocqueues_iwrite, wait_iwrite
 
-    rocqueues  = Array{AMDGPU.HSAQueue}(undef, NNEIGHBORS_PER_DIM, 0)
-    rocsignals = Array{Union{AMDGPU.HSASignal,AMDGPU.RuntimeEvent{AMDGPU.HSAStatusSignal},Missing}}(undef, NNEIGHBORS_PER_DIM, 0)
+    rocqueues  = Array{AMDGPU.ROCQueue}(undef, NNEIGHBORS_PER_DIM, 0)
+    rocsignals = Array{Union{AMDGPU.ROCSignal,AMDGPU.ROCKernelSignal,Missing}}(undef, NNEIGHBORS_PER_DIM, 0)
```

Comment:
@jpsamaroo can you maybe have a look at the implementation of AMDGPU here to see if we overlooked something wrt the latest v0.4.0 changes?